File: DemoServer/AdditionalLanguages/python/queries/queryExample/queryExample.py
from typing import Optional, List
from demo_example import Example
from models import Employee

#region Usings
from ravendb import QueryStatistics
#endregion

class QueryExample(Example):
    def run(self, run_params=None) -> List[Employee]:
        #region Demo
        with self.document_store_holder.store().open_session() as session:
            stats: Optional[QueryStatistics] = None

            def __stats_callback(statistics: QueryStatistics):
                nonlocal stats
                stats = statistics

            #region Step_1
            query = session.query(object_type=Employee)
            #endregion
            #region Step_2
            query = query.where_equals("FirstName", "Steven")\
                .or_else()\
                .where_equals("Title", "Sales Representative")
            #endregion
            #region Step_3
            query = query.include("ReportsTo")
            #endregion
            #region Step_4
            query = query.statistics(__stats_callback)
            #endregion
            #region Step_5
            query = query.order_by_descending("HiredAt")
            #endregion
            #region Step_6
            query = query.take(5)
            #endregion

            #region Step_7
            query_results = list(query)
            #endregion
        #endregion

        return query_results

File: DemoServer/AdditionalLanguages/php/queries/queryExample/QueryExample.php
<?php

namespace RavenDB\Demo\queries\queryExample;

//region Usings
use RavenDB\Documents\Session\QueryStatistics;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Employee;

class QueryExample
{
    public function __invoke(): array
    {
        //region Demo
        $queryResults = [];

        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            $statisticsRef = new QueryStatistics();

            //region Step_1
            $query = $session->query(Employee::class)
            //endregion
                //region Step_2
                ->whereEquals("FirstName", "Steven")
                ->orElse()
                ->whereEquals("Title", "Sales Representative")
                //endregion
                //region Step_3
                ->include("ReportsTo")
                //endregion
                //region Step_4
                ->statistics($statisticsRef)
                //endregion
                //region Step_5
                ->orderByDescending("HiredAt")
                //endregion
                //region Step_6
                ->take(5);
                //endregion

            //region Step_7
            $queryResults = $query->toList();
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $queryResults;
    }
}
